voluncheer/opportunityboard/views/search.py

```diff
@@ -65,10 +66,32 @@ def search(self):
             filtered_opportunity = filtered_opportunity.filter(subcategory=self.subcategory)
         if self.subsubcategory is not None:
             filtered_opportunity = filtered_opportunity.filter(subsubcategory=self.subsubcategory)
-
+        if self.duration == "0-2 Hours":
+            filtered_opportunity = filter_by_duration(filtered_opportunity, 0, 2)
+        if self.duration == "2-4 Hours":
+            filtered_opportunity = filter_by_duration(filtered_opportunity, 2, 4)
+        if self.duration == "4-8 Hours":
+            filtered_opportunity = filter_by_duration(filtered_opportunity, 4, 8)
+        if self.duration == "8+ Hours":
+            filtered_opportunity = filter_by_duration(filtered_opportunity, 8, 24)
         return filtered_opportunity
 
 
+def filter_by_duration(queryset, min, max):
+    """
+    Takes Opportunity queryset, as well as a min and a max number of hours
+    Returns Opportunity queryset with durations only between min and max hours
+    """
+    min_duration = dt.timedelta(hours=min)
+    max_duration = dt.timedelta(hours=max)
+    for opportunity in queryset:
```

Comment:
An alternative would be to accumulate the opportunities that match the criteria via something like list.append instead of excluding one's that do not match. It's mostly a personal choice but I generally find inclusion easier to read that exclusion. Up to you though.

Reply:
My thought process was that we already have a queryset filtered by category without regard to duration.  It feels more intuitive to me to further hone down the existing queryset by removing opportunities that are too long, rather than creating a new queryset that appends items from the existing one, at least given the option of Django's `filter ` vs. `exclude`.